internal/db/schema/migrations/oss/postgres/56/03_create_kms_key_destructions.up.sql

```diff
@@ -18,7 +18,8 @@ begin;
     key_id kms_private_id not null
       references kms_data_key_version_destruction_job (key_id)
         on delete cascade
-        on update cascade,
+        on update cascade
+        deferrable initially deferred,
```

Comment:
This lets us try to create the job_runs before the job, which means we don't have to create the job and then delete it if there were no runs.

Reply:
It happens in the same transaction as the job runs in `DestroyKeyVersion`: https://github.com/hashicorp/boundary/pull/2498/files#diff-65fa95abdd78accdc6b1f3ac98751fe4700c8abc1ee696b8d6ae7e5403af0831R504. What do you mean that they will be deleted? It will error at commit and rollback if these constraints are not met, it won't delete anything silently.